### PR TITLE
Add block type

### PR DESCRIPTION
**Pull request by saltz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
   "scripts": {
     "build": "tsc -p tsconfig.json",
     "build-watch": "tsc -w -p tsconfig.json",
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "lint": "tslint --project tsconfig.json"
+    "lint": "tslint --project tsconfig.json",
+    "dev": "npm run build && npm run lint"
```
Comment (roelvdboom) [inline package.json]: order alphabetically

File: src/block.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
  * Default block in the chain
  */
 export class Block implements IModel {
+  public blockType: BlockType;
   public blockHeader: BlockHeader;
   public transactions: Set<Transaction>;
 
-  constructor(blockHeader: BlockHeader, transactions: Set<Transaction>) {
+  constructor(blockHeader: BlockHeader, transactions: Set<Transaction>, blockType?: BlockType) {
+    this.blockType = blockType || BlockType.REGUALAR;
```
Comment (roelvdboom) [inline src/block.ts]: `REGUALAR` ?????

File: src/blockType.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Defines the type of the block
+ */
+export enum BlockType {
+    GENESIS,
+    REGUALAR,
```
Comment (roelvdboom) [inline src/blockType.ts]: Same spelling error

File: src/block.ts [hunk 4]
```diff
@@ -1,15 +1,18 @@
-import { BlockHeader } from "./blockHeader";
+import {BlockHeader} from "./blockHeader";
```
Comment (roelvdboom) [inline src/block.ts]: add spaces

Comment (TimoF97) [inline src/block.ts]: I changed this inconsistency in my PR yesterday. Please make sure to change the import to `import { BlockHeader } from`.

File: src/blockType.ts (hunk 3 above)
Comment (TimoF97) [inline src/blockType.ts]: Typo. Please change to `REGULAR`

File: src/block.ts (hunk 4 above)
Comment (roelvdboom) [inline src/block.ts]:
```suggestion
import { BlockHeader } from "./blockHeader";
```

File: src/block.ts (hunk 2 above)
Comment (roelvdboom) [inline src/block.ts]:
```suggestion
    this.blockType = blockType || BlockType.REGULAR;
```

File: src/blockType.ts (hunk 3 above)
Comment (roelvdboom) [inline src/blockType.ts]:
```suggestion
    REGULAR,
```

File: src/block.ts (hunk 2 above)
Comment (TimoF97) [reply src/block.ts]: Please consider changing the optional parameter to a default paramater value.

i.e.
```
constructor(blockHeader: BlockHeader, transactions: Set<Transaction>, blockType: BlockType = BlockType.REGUALAR) {
    this.blockType = blockType;
```

Comment (saltz) [reply src/block.ts]: I thought we discussed to make it nullable and resolve it to the default value: REGULAR.
This way a block gets the type GENISIS if explicitly stated. This way fewer parameters need to be passed in the constructor.

**Pull request merged**